src/app/components/core/control-dispatcher/add-control-acl/add-control-acl.component.ts

```diff
@@ -0,0 +1,307 @@
+import { ActivatedRoute, Router } from '@angular/router';
+import { AlertService, ProgressBarService, ServicesApiService, SharedService } from '../../../../services';
+import { ControlDispatcherService } from '../../../../services/control-dispatcher.service';
+import { DialogService } from '../confirmation-dialog/dialog.service';
+import { uniqBy } from 'lodash';
+
+@Component({
+  selector: 'app-add-control-acl',
+  templateUrl: './add-control-acl.component.html',
+  styleUrls: ['./add-control-acl.component.css']
+})
+export class AddControlAclComponent implements OnInit {
+  services = [];
+  serviceTypes = []
+  aclURLsList = [];
+
+  aclServiceTypeList = [];
+  serviceNameList = [];
+  serviceTypeList = [];
+  name: string;
+  update = false;
+
+  @ViewChild('aclForm') aclForm: NgForm;
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private controlService: ControlDispatcherService,
+    private alertService: AlertService,
+    private ngProgress: ProgressBarService,
+    private dialogService: DialogService,
+    private servicesApiService: ServicesApiService,
+    public sharedService: SharedService) { }
+
+  ngOnInit(): void {
+    this.route.params.subscribe(params => {
+      this.name = params['name'];
+      if (this.name) {
+        this.update = true;
+        this.getACLbyName();
+      }
+    });
+    this.getAllServices();
+  }
+
+  ngAfterViewInit() {
+    setTimeout(() => {
+      if (!this.update) {
+        this.addFormControls(0);
+      }
+    }, 0);
+  }
+
+  public toggleDropDown(id: string) {
+    const activeDropDowns = Array.prototype.slice.call(document.querySelectorAll('.dropdown.is-active'));
+    if (activeDropDowns.length > 0) {
+      if (activeDropDowns[0].id !== id) {
+        activeDropDowns[0].classList.remove('is-active');
+      }
+    }
+    const dropDown = document.querySelector(`#${id}`);
+    dropDown.classList.toggle('is-active');
+  }
+
+  openModal(id: string) {
+    this.dialogService.open(id);
+  }
+
+  closeModal(id: string) {
+    this.dialogService.close(id);
+  }
+
+  getACLbyName() {
+    /** request started */
+    this.ngProgress.start();
+    this.controlService.fetchAclByName(this.name)
+      .subscribe((res: any) => {
+        this.ngProgress.done();
+        this.serviceNameList = res.service.filter(item => item.name);
+        this.serviceTypeList = res.service.filter(s => s.type);
+        this.serviceTypeList = uniqBy(this.serviceTypeList, 'type');
+        this.aclURLsList = [];
+        this.aclURLsList = res.url;
+        this.aclServiceTypeList = [];
+        this.aclURLsList.forEach((item, index) => {
+          item.index = index;
+          if (item.acl) {
+            item.acl.forEach(acl => {
+              this.aclServiceTypeList.push({ type: acl.type, index });
+            });
+            this.addNewURLControl(index, this.aclURLsList[index]);
+          }
+        });
```

Comment:
please add comment what each internal item is for, or name them more explanatory